src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2895,6 +2897,451 @@ public void Inheritance_17()
+using System;
+
+data class B
+{
+    public int X() => 3;
+}
+
+data class C(int X, int Y) : B
+{
+    public new int X { get; }
+
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C(int x, int y):
+                Console.Write(x);
+                Console.Write(y);
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new C(1, 2));
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: "02");
+            verifier.VerifyDiagnostics();
+
+            var cClass = ((CSharpCompilation)verifier.Compilation).GetMember<NamedTypeSymbol>("C");
+            var deconstruct = cClass.GetMember("Deconstruct");
+            Assert.NotNull(deconstruct);
+        }
+
+        [Fact]
+        public void Deconstruct_MethodCollision_04()
+        {
+            var source = @"
+data class C(int X, int Y)
+{
+    public int X(int arg) => 3;
+
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C(int x, int y):
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new C(1, 2));
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (2,18): error CS0102: The type 'C' already contains a definition for 'X'
+                // data class C(int X, int Y)
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "X").WithArguments("C", "X").WithLocation(2, 18));
+
+            var cClass = comp.GetMember<NamedTypeSymbol>("C");
+            var deconstruct = cClass.GetMember("Deconstruct");
+            Assert.NotNull(deconstruct);
+        }
+
+        [Fact]
+        public void Deconstruct_Empty()
+        {
+            var source = @"
+data class C
+{
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C():
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new C());
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,19): error CS1061: 'C' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'C' could be found (are you missing a using directive or an assembly reference?)
+                //             case C():
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "()").WithArguments("C", "Deconstruct").WithLocation(8, 19),
+                // (8,19): error CS8129: No suitable 'Deconstruct' instance or extension method was found for type 'C', with 0 out parameters and a void return type.
+                //             case C():
+                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "()").WithArguments("C", "0").WithLocation(8, 19));
+
+            var cClass = comp.GetMember<NamedTypeSymbol>("C");
+            var deconstruct = cClass.GetMember("Deconstruct");
+            Assert.Null(deconstruct);
+        }
```

Comment:
Please test a derived type that does not add parameters:
```C#
data class A(object X, object Y);
data class B(object X, object Y) : A;
```

Reply:
Added in `Deconstruct_Inheritance_01`

Reply:
`Deconstruct_Inheritance_01` does not include the base properties on the derived record.

Reply:
Sorry, now I think I get it, does `Deconstruct_Inheritance_03` look like what you wanted?